homeassistant/components/vizio/media_player.py

```diff
@@ -330,17 +330,25 @@ async def async_added_to_hass(self) -> None:
             )
         )
 
+        if not self._apps_coordinator:
+            return
+
         # Register callback for app list updates if device is a TV
         @callback
-        def apps_list_update():
+        def apps_list_update() -> None:
             """Update list of all apps."""
+            # If the data hasn't changed we don't need to do a state update
+            if (
+                not self._apps_coordinator
+                or self._all_apps == self._apps_coordinator.data
+            ):
+                return
```

Comment:
The state engine already does this for you. If the state didn't change, nothing happens. So there should be no need to do this yourself as well.

Reply:
sorry, this isn't the state engine though, this is the coordinator. Does that also silently not fire if the data is the same?